### PR TITLE
refactor(client): remove as_pandas parameter

### LINKED ISSUE #1360: [Client] Move to 'rb.DatasetFor*' by default
- Remove `as_pandas` argument in the `rb.load` method, return a `rb.DatasetFor*` by default
- Return `rb.DatasetFor*` objects instead of records `list`s in the `labeling` module

### PR DESCRIPTION
**Pull request by frascuchon**
Closes #1360

### CODE REVIEW

**Comment by dcfidalgo** [comment]
@frascuchon Should we include the adaptations of the docs in this PR, or do this in a follow-up PR?

**Comment by frascuchon** [comment]
Yes, I forget the docs. Do you have located how many docs are affected by this change?

**Comment by dcfidalgo** [comment]
Not really, but I can take care of this in this PR.

**Comment by dcfidalgo** [comment]
@frascuchon I updated the docu part. I also introduced a FutureWarning instead of completely removing the parameter in order not to break code. At the moment I think there some users that have `rb.load(..., as_pandas=False)` in their codebase, so I think we should notify them first, so that they can remove this argument, before breaking their code.

**Review by dcfidalgo — DISMISSED** [review]

File: src/rubrix/client/api.py [hunk 1]
```diff
@@ -401,26 +400,32 @@ def load(
...
-            >>> dataframe = rb.load(name="example-dataset")
             >>> dataset = rb.load(name="example-dataset")
         """
+        if as_pandas is not None:
+            warnings.warn(
+                "The argument `as_pandas` is deprecated and will be removed in a future version. "
+                "Please adapt your code accordingly. "
+                "If you want a pandas DataFrame do `rb.load(...).to_pandas()`."
+            )
+
```
Comment (frascuchon) [inline src/rubrix/client/api.py]: This method will work if `as_pandas=False` but the user flow will fail when passing `as_pandas=True`. Also, the error can be confusing for the user.

Maybe, we should include a check here to raise a `ValueError` when `as_pandas=True` 

```suggestion
        if as_pandas is False:
            warnings.warn(
                "The argument `as_pandas` is deprecated and will be removed in a future version. "
                "Please adapt your code accordingly. "
            )
        elif as_pandas is True:
           raise ValueError(
                "If you want a pandas DataFrame do `rb.load(...).to_pandas()`."
            )

```

Comment (dcfidalgo) [reply src/rubrix/client/api.py]: :+1: Done, I just made the error message a bit more explicit.

**Review by dcfidalgo — APPROVED** [review]

**Comment by dcfidalgo** [comment]
From my side, this is good to go, merge it at will.

**Pull request merged**